orttraining/orttraining/training_ops/cuda/tensor/split.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "split.h"
+#include "core/providers/cuda/tensor/split_impl.h"
+#include "core/providers/cpu/tensor/utils.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+namespace cuda {
+ONNX_OPERATOR_KERNEL_EX(SplitTraining,
+                        kMSDomain,
+                        1,
+                        kCudaExecutionProvider,
+                        KernelDefBuilder()
+                            .InputMemoryType<OrtMemTypeCPUInput>(1)
```

Comment:
In this case the second input would come from 'Concat' which is implemented in cuda provider as well. Shouldnt restrict to CPU memory

Reply:
if I don't add this, the UT would fail at #78 below. 

---
In reply to: [459769176](https://github.com/microsoft/onnxruntime/pull/4597#discussion_r459769176) [](ancestors = 459769176)